src/producer.rs

```diff
@@ -78,9 +79,10 @@ impl FileProducer {
 
 impl Producer for FileProducer {
   fn produce(&mut self) -> ProducerState<&[u8]> {
+    let len = self.v.len();
     //let mut v = Vec::with_capacity(self.size);
     //self.v.clear();
-    self.v.resize(self.size, 0);
+    self.v.extend(repeat(0).take(self.size - len));
```

Comment:
I'm making the assumption here that `resize` only every grows the vector. I might be mistaken there - I only glanced at the surroundings.